Formula/bzip3.rb

```diff
@@ -0,0 +1,30 @@
+class Bzip3 < Formula
+  desc "Better and stronger spiritual successor to BZip2"
+  homepage "https://github.com/kspalaiologos/bzip3"
+  url "https://github.com/kspalaiologos/bzip3/releases/download/1.2.1/bzip3-1.2.1.tar.gz"
+  sha256 "bba90d867f53efa4291de1df9c22da1578e1a93ca819e0fc68881da6fcc4149d"
+  license "LGPL-3.0-only"
+
+  # Fix -flat_namespace being used on Big Sur and later.
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/03cf8088210822aa2c1ab544ed58ea04c897d9c4/libtool/configure-big_sur.diff"
+    sha256 "35acd6aebc19843f1a2b3a63e880baceb0f5278ab1ace661e57a502d9d78c93c"
+  end
```

Comment:
can we also PR this change to the bzip3 repo?

Reply:
That is unclear to me, I discussed it upstream here:

https://github.com/kspalaiologos/bzip3/issues/81

They are already using autoconf 2.71 which is the latest version, so I don't quite understand if there is an unreleased autoconf that fixes the flat namespace configure script issue. I'm not sure they want to maintain a patch diff and we have ones already that are used by 264 formulae currently.

Reply:
Those patches are only needed if the configure script hasn't been regenerated with a recent autoconf version, which not all older software is. We hold ourselves to higher standards for new additions though.

Reply:
Do you know what version of autoconf fixes this issue? They are on 2.71 and still have the same configure script generated that patch is for?

Reply:
Our 2.71 doesn't have any patches and doesn't generate configure files that have this problem. Maybe it's an Ubuntu thing?

Reply:
I will work with the upstream bzip3 project to investigate and fix this.

https://github.com/kspalaiologos/bzip3/issues/81

Is there consideration to have this formula merged, or do you desire to wait for an upstream fix and a future new upstream release?

Reply:
if we have the PR for the upstream, I think we can go with the merge (otherwise, the patch might persist forever)

Reply:
It looks like the issue is libtool on ubuntu 22.04 LTS is version 2.4.6 which is the latest stable release, and contains the issue that requires patching.

https://www.gnu.org/software/libtool/

I suppose when libtool releases their next stable version (it's fixed in their git repo for the current development branch), and it propagates to the next ubuntu LTS, this will be resolved?

Considering the patch works against current libtool 2.4.6 stable release, I suggest merging the formula now. Requiring upstream to use a development version of libtool seems a little over the top?

Reply:
Actually @chenrui333 the bzip3 maintainer is super responsive and has made a commit that integrates the patch to resolve this issue for macOS at least until his release builders use libtool 2.4.7 and greater.

https://github.com/kspalaiologos/bzip3/commit/e667cacfaa0b4b56e45d48e0496041c376c82d53

It will be included in the next release version of bzip3 1.2.2 whenever it is made.

Can we merge this?

Reply:
Yeah, totally, just added a comment about the upstream commit reference.